api/all/src/main/java/io/opentelemetry/api/trace/Span.java

```diff
@@ -22,6 +22,14 @@
  *
  * <p>Spans are created by the {@link SpanBuilder#startSpan} method.
  *
+ * <p>When inserting attributes, some implementations may prohibit {@code null} keys and values, and
```

Comment:
"some implementations may prohibit" is wrong IMHO. The API should already prohibit this (as does the spec). It might be better to say "null keys and values are not allowed and the behavior is undefined. Some implementations might be able to cope with them, e.g. by ignoring them while others might throw exceptions (e.g. NullPointerExceptions) or an error might occur later in an exporter."

Reply:
This brings up an excellent point, @Oberon00 . If the spec "prohibits" this, but the spec also requires that mis-use of the APIs should never throw exceptions, how are we supposed to implement these two conflicting requirements, in general?

Reply:
The compliance matrix has an entry "`null` values documented as invalid/undefined"

I don't see these requirements as conflicting: On the API level, the behavior is undefined. But for the two concrete implementations in the spec (SDK and Noop-implementation) the spec places the additional requirement on them to not throw errors / return null. However, we should not advertise this to users as behavior they can rely on.

Reply:
I like the teminology `"null keys and values are not allowed and the behavior is undefined.` at this top level. And I don't think we need any other docs since we don't have any `@Nulable` annotations on the methods themselves - passing `null` would mean they ignored the docs anyways :)

Reply:
If I understood correctly the suggestion this should be done.